### PR TITLE
[MAINT] update parameters for `scipy.sparse.linalg.cg`

### LINKED ISSUE #4396: :rotating_light: NIGHTLY DEPENDENCIES TEST: failure on `refs/heads/main`
The run `8815256231` of the workflow testing Nilearn with the nightly build of its dependencies failed on `refs/heads/main`.

You can view the report here: https://github.com/nilearn/nilearn/actions/runs/8815256231

Pinging @nilearn/core

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4396

See https://github.com/nilearn/nilearn/actions/runs/8797817809/job/24143526441#step:7:26389

```
FAILED nilearn/_utils/tests/test_segmentation.py::test_modes_in_random_walker - TypeError: cg() got an unexpected keyword argument 'tol'
FAILED nilearn/_utils/tests/test_segmentation.py::test_isolated_seed - TypeError: cg() got an unexpected keyword argument 'tol'
FAILED nilearn/_utils/tests/test_segmentation.py::test_reorder_labels - TypeError: cg() got an unexpected keyword argument 'tol'
FAILED nilearn/regions/tests/test_region_extractor.py::test_connected_regions_4D[local_regions] - TypeError: cg() got an unexpected keyword argument 'tol'
FAILED nilearn/regions/tests/test_region_extractor.py::test_connected_regions_3D[local_regions] - TypeError: cg() got an unexpected keyword argument 'tol'
FAILED nilearn/regions/tests/test_region_extractor.py::test_connected_regions_different_results_with_different_mask_images - TypeError: cg() got an unexpected keyword argument 'tol'
FAILED nilearn/regions/tests/test_region_extractor.py::test_region_extractor_fit_and_transform - TypeError: cg() got an unexpected keyword argument 'tol'
FAILED nilearn/regions/tests/test_region_extractor.py::test_region_extractor_strategy_ratio_n_voxels - TypeError: cg() got an unexpected keyword argument 'tol'
FAILED nilearn/regions/tests/test_region_extractor.py::test_region_extractor_strategy_percentile - TypeError: cg() got an unexpected keyword argument 'tol'
FAILED nilearn/regions/tests/test_region_extractor.py::test_region_extractor_high_resolution_image - TypeError: cg() got an unexpected keyword argument 'tol'
FAILED nilearn/regions/tests/test_region_extractor.py::test_region_extractor_zeros_affine_diagonal - TypeError: cg() got an unexpected keyword argument 'tol'
```

Changes proposed in this pull request:

- The `tol` argument of scipy.sparse.linalg.cg is deprecated in favour of `rtol` and will be removed in SciPy 1.14. Furthermore, the default value of `atol` for these functions is due to change to `0.0` in SciPy 1.14.

See https://docs.scipy.org/doc/scipy/release/1.12.0-notes.html#deprecated-features

### CODE REVIEW

**Comment by Remi-Gau** [comment]
failures on python 3.8: it seems that with python 3.8 will, we can only install scipy 1.10

File: nilearn/_utils/segmentation.py [hunk 1]
```diff
@@ -355,7 +357,14 @@ def _solve_cg(lap_sparse, B, tol):
     lap_sparse = lap_sparse.tocsc()
     X = []
     for i in range(len(B)):
-        x0 = cg(lap_sparse, -B[i].todense(), tol=tol, atol="legacy")[0]
+        # TODO Python 3.8
+        # consider remove if/else block when dropping support for 3.8
+        # would require pinning scipy to >= 1.12
+        # See https://github.com/nilearn/nilearn/pull/4394
```
Comment (Remi-Gau) [inline nilearn/_utils/segmentation.py:363]: Will open an issue after merging to keep track of the things to do when we drop python 3.8.

**Review by man-shu — APPROVED** [review]
Looks good! 
The codecov annotation is wrong, the coverage is actually 100%.
Screenshot 2024-04-24 at 10 17 34 AM

**Review by bthirion — APPROVED** [review]
LGTM

**Pull request merged**